### PR TITLE
Update heffte to 2.2.0

### PR DESCRIPTION
**Pull request by sfogerty**
Update heffte version to 2.2.0

needed a slight change to the `alltoall` option. We had that as a bool and heffte now has three options. Future change should allow choosing any of those three options, but for now it's true -> all-to-all, and false -> point-to-point. (heffte also allows p2p_plined)

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
       - name: Checkout heffte
         # actions/checkout doesn't work for external repos yet
         run: |
-          git clone --depth 1 --branch v2.0.0 https://bitbucket.org/icl/heffte.git heffte
+          git clone --depth 1 --branch v2.2.0 https://bitbucket.org/icl/heffte.git heffte
```
Comment (dalg24) [inline .github/workflows/docker.yml:37]: Was that used at all?  We were requiring v2.1.0 so I am not sure what was going on.

Comment (sfogerty) [reply .github/workflows/docker.yml:37]: I am not sure that Cabana_REQUIRE_HEFFTE is being set in the CI everywhere it should be

Comment (streeve) [reply .github/workflows/docker.yml:37]: You may be right, but @junghans any idea why all the Cajita runs are failing on the docker build?

**Comment by streeve** [comment]
Closing in favor of #538

**Pull request closed without merging**